lib/bus.rb

```diff
@@ -25,11 +34,15 @@ def self.[](key)
   end
 
   def self.[]=(key, value)
-    redis.pipelined do
-      redis.set(key, value)
-      redis.expire(key, TTL)
+    redis.pipelined do |pipeline|
+      pipeline.set(key, value)
+      pipeline.expire(key, ttl_for(key))
     end
 
     value
   end
+
+  def self.ttl_for(key)
+    TTL.find { |k, _v| key.include?(k) }&.last || 0
```

Comment:
why not just TTL[key] is key some kind of dict?

Reply:
Key has a unique id in it. Example, if the user's id is 1234, the key would be "USER_STATS:1234"

Reply:
this should be explicit, get the key by splitting the string and then doing normal hash lookip

Reply:
Fixed.